### PR TITLE
Add camera PTZ article

### PR DESCRIPTION
**Pull request by beaufortfrancois**
Fixes #https://github.com/GoogleChrome/web.dev/issues/3723

Changes proposed in this pull request:

- Add the Camera PTZ article.
- Add Camera PTZ to the "New capabilities status" page.

Live preview: 
- https://deploy-preview-3923--web-dev-staging.netlify.app/ptz/
- https://deploy-preview-3923--web-dev-staging.netlify.app/fugu-status/

### CODE REVIEW

**Comment by tomayac** [comment]
Meta: consider renaming the article folder to have a "speaking" URL: `s/ptz/camera-pan-tilt-zoom`.

File: src/site/content/en/blog/ptz/index.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+- [ChromeStatus entry](https://www.chromestatus.com/feature/5570717087170560)
+- [Chrome tracking bug](https://bugs.chromium.org/p/chromium/issues/detail?id=934063)
+
+## Acknowledgements
+
+Thanks to [Joe Medley] for his review of this article. Hero image by [Christina
+@ wocintechchat.com] on [Unsplash].
+
+<!-- lint disable definition-case -->
+[mandatory constraints]: https://developer.mozilla.org/en-US/docs/Web/API/Media_Streams_API/Constraints
```
Comment (tomayac) [inline src/site/content/en/blog/ptz/index.md:217]:
```suggestion
[mandatory constraints]: https://developer.mozilla.org/en-US/docs/Web/API/Media_Streams_API/Constraints#Specifying_a_range_of_values:~:text=Constraints%20which%20are%20specified%20using%20any,applyConstraints()%2C%20the%20promise%20will%20be%20rejected.
```

Better deeplink.

File: src/site/content/en/blog/ptz/index.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+## Playground  {: #playground }
+
+You can play with the API by running the [demo] on Glitch. Be sure to [check out
+the source code].
+
+Tip: If you don't have a camera that supports PTZ, you can [run Chrome with the
+switch] `--use-fake-device-for-media-stream` to simulate one on your machine.
+Enjoy!
+
+{% Glitch {
```
Comment (tomayac) [inline src/site/content/en/blog/ptz/index.md:190]: You make use of `color-scheme: dark light` (🎉), but the iframe doesn't respect it, only the standalone doc. You need to add…

```css
@media (prefers-color-scheme: dark) {
  body {
    color: #fff; 
    background-color: #000;
  }
}
```

…to your CSS.

**Review by tomayac — APPROVED** [review]
LGTM, approval, with nits.

File: src/site/content/en/blog/ptz/index.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+does not automatically gain PTZ access if it becomes available. This is true
+even when the camera itself supports PTZ. The permission must be requested
+again. Fortunately, you can use the [Permissions API] to query and monitor the
+status of PTZ permission.
+
+```js
+try {
+  const panTiltZoomPermissionStatus = await navigator.permissions.query({
+    name: "camera",
+    panTiltZoom: true
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:90]: They actually can be queried **and** requested separately with the Permissions API.

File: src/site/content/en/blog/ptz/index.md (hunk 1 above)
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:217]: Thx!

File: src/site/content/en/blog/ptz/index.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+  - blog # blog is a required tag for the article to show up in the blog.
+  - capabilities
+  - media
+feedback:
+  - api
+---
+
+Room-scale video conferencing solutions deploy cameras with pan, tilt, and zoom
+(PTZ) capabilities so that software can point the camera at meeting
+participants. Starting in Chrome 87, the pan, tilt, and zoom features on camera
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:23]: Done.

File: src/site/content/en/blog/ptz/index.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+## Using the API {: #use }
+
+### Feature detection {: #feature-detection }
+
+Feature detection for hardware is different from what you're probably used to.
+The presence of `"pan"`, `"tilt"`, and `"zoom"` constraint names in
+`navigator.mediaDevices.getSupportedConstraints()` tells you that the browser
+supports the API to control camera PTZ, but not whether the camera hardware
+supports it. As of Chrome 87, controlling camera PTZ is supported on Android
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:35]: Done

File: src/site/content/en/blog/ptz/index.md (hunk 2 above)
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:190]: As discussed offline, this is not enough. I think we should add proper dark theme support to all web.dev pages.

File: src/site/content/en/blog/ptz/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/ptz/index.md:90]: So to query if I currently have the permission to use the camera exclusively with zoom, I could do that like so?

```js
const panTiltZoomPermissionStatus = await navigator.permissions.query({
    name: "camera",
    pan: false,
    tilt: false,
    zoom: true,
});
```

[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:90]: Once again (my bad) I misunderstood. You're right.
The PTZ permission is a separate one from camera that includes pan, tilt, and zoom camera features.
From user perspective, website can access PTZ features or not, hence the PTZ permission.
From web dev perspective, we've reused existing media track constraints `pan`, `tilt`, and `zoom` to detect whether or not the PTZ permission would be requested.

**Comment by jpmedley** [comment]
Doing fail-safe check now.

File: src/site/content/en/blog/ptz/index.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+- [Chrome tracking bug](https://bugs.chromium.org/p/chromium/issues/detail?id=934063)
+
+## Acknowledgements
+
+Thanks to [Joe Medley] for his review of this article. Hero image by [Christina
+@ wocintechchat.com] on [Unsplash].
+
+<!-- lint disable definition-case -->
+[mandatory constraints]: https://developer.mozilla.org/en-US/docs/Web/API/Media_Streams_API/Constraints#Specifying_a_range_of_values:~:text=mandatory
+[`MediaStream`]: https://w3c.github.io/mediacapture-main/#stream-api
```
Comment (jpmedley) [inline src/site/content/en/blog/ptz/index.md:219]:
```suggestion
[`MediaStream`]: https://developer.mozilla.org/en-US/docs/Web/API/MediaStream
```

**Review by jpmedley — CHANGES_REQUESTED** [review]
There are all pretty simple changes. I'll be around tomorrow to approve as soon as you're done.

File: src/site/content/en/blog/fugu-status/index.md [hunk 7]
```diff
@@ -329,6 +329,16 @@ latest version of Chrome, and in many cases other Chromium based browsers.
           <em>Updated November 8, 2019</em>
         </td>
       </tr>
+      <tr>
+        <td>
+          <a href="/ptz/">Camera pan, tilt, and zoom</a>
+        </td>
+        <td>
+          Pan, tilt, and zoom features on cameras are finally accessible on the
+          web.<br>
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/fugu-status/index.md:338]: Done

File: src/site/content/en/blog/ptz/index.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+  - capabilities
+  - media
+feedback:
+  - api
+---
+
+Room-scale video conferencing solutions deploy cameras with pan, tilt, and zoom
+(PTZ) capabilities so that software can point the camera at meeting
+participants. Starting in Chrome&nbsp;87, the pan, tilt, and zoom features on
+camera are accessible to websites using media track constraints in
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:24]: Done

File: src/site/content/en/blog/ptz/index.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+## Using the API {: #use }
+
+### Feature detection {: #feature-detection }
+
+Feature detection for hardware is different from what you're probably used to.
+The presence of `"pan"`, `"tilt"`, and `"zoom"` constraint names in
+`navigator.mediaDevices.getSupportedConstraints()` tells you that the browser
+supports the API to control camera PTZ, but not whether the camera hardware
+supports it. As of Chrome&nbsp;87, controlling camera PTZ is supported on
+Android (zoom only) and desktop platforms.
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:36]: No problem.

File: src/site/content/en/blog/ptz/index.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+  <img src="./media-internals-page.jpg" class="w-screenshot" alt="Screenshot of the internal page in Chrome OS to debug PTZ camera support.">
+  <figcaption class="w-figcaption">Internal page to debug PTZ camera support.</figcaption>
+</figure>
+
+### Control camera PTZ {: #control }
+
+Manipulate camera PTZ capabilities and settings using the preview
+`MediaStreamTrack` from the `stream` object obtained earlier.
+`MediaStreamTrack.getCapabilities()` returns a dictionary with the supported
+capabilities and the ranges or allowed values. Correspondingly,
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:120]: MediaTrackCapabilities can return either ranges (min, max, step) or allowed values. , e.g. supported zoom range or allowed white balance modes.  
See https://w3c.github.io/mediacapture-image/#mediatrackcapabilities-section

File: src/site/content/en/blog/ptz/index.md (hunk 6 above)
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:219]: Done

File: src/site/content/en/blog/ptz/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/ptz/index.md:91]: We're discussing offline.

**Comment by beaufortfrancois** [comment]
> There are all pretty simple changes. I'll be around tomorrow to approve as soon as you're done.

Thanks. Let's not merge now though.

File: src/site/content/en/blog/ptz/index.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/ptz/index.md:120]: Hm. The IDL seems out of date. https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/modules/mediastream/media_track_capabilities.idl

**Review by jpmedley — APPROVED** [review]

**Comment by beaufortfrancois** [comment]
@jpmedley This PR was NOT meant to be merged. I've explicitly told you at  https://github.com/GoogleChrome/web.dev/pull/3923#issuecomment-698149211 and added the `DO NOT MERGE" label.
 
image
 
This feature is for Chrome 87 and we're not sure yet when we're going to launch it as communicated offline earlier as well.

**Comment by beaufortfrancois** [comment]
I've re-opened a new PR for this article at https://github.com/GoogleChrome/web.dev/pull/3923. 
@jpmedley Please approve and I will take care of merging.

**Comment by jpmedley** [comment]
I am so sorry.

**Pull request merged**